pkg/k8s/resource_printer.go

```diff
@@ -61,19 +68,38 @@ func (y *YAMLObjectPrinter) PrintObject(obj runtime.Object, w io.Writer) error {
 
 type JSONObjectPrinter struct{}
 
-func (j JSONObjectPrinter) PrintObject(obj runtime.Object, w io.Writer) error {
-	data, err := json.Marshal(obj)
-	if err != nil {
-		return err
+func (j JSONObjectPrinter) PrintObject(objs []runtime.Object, w io.Writer) error {
+	base := new(bytes.Buffer)
+
+	if len(objs) > 1 {
+		base.WriteRune('[')
 	}
 
-	var buf bytes.Buffer
-	err = json.Indent(&buf, data, "", "    ")
-	if err != nil {
-		return err
+	for _, obj := range objs {
```

Comment:
Is this change needed? `json.Marshal` should automatically add the start and ending brackets when passed in a slice.

Reply:
Each object is json but the whole thing is not so if you try to marshal it, it fails